iothub/device/src/Common/Security/SharedAccessSignature.cs

```diff
@@ -116,7 +117,7 @@ public void Authenticate(SharedAccessSignatureAuthorizationRule sasAuthorization
             if (sasAuthorizationRule.PrimaryKey != null)
             {
                 string primareyKeyComputedSignature = ComputeSignature(Convert.FromBase64String(sasAuthorizationRule.PrimaryKey));
-                if (StringComparer.Ordinal.Equals(Signature, primareyKeyComputedSignature))
+                if (FixedTimeEquals(Signature, primareyKeyComputedSignature))
```

Comment:
https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.cryptographicoperations.fixedtimeequals?view=net-5.0 is available for ns2.1+, so an if-def for lower target frameworks would work here.